src/com/oltpbenchmark/schema/defaultschema/DefaultSchemaManager.java

```diff
@@ -0,0 +1,121 @@
+package com.oltpbenchmark.schema.defaultschema;
+
+import org.apache.log4j.Logger;
+
+import com.oltpbenchmark.api.BenchmarkModule;
+import com.oltpbenchmark.benchmarks.tpcc.procedures.StockLevel;
+import com.oltpbenchmark.schema.SchemaManager;
+import com.oltpbenchmark.schema.TPCCTableSchemas;
+import com.oltpbenchmark.schema.Table;
+import com.oltpbenchmark.schema.TableSchema;
+import com.oltpbenchmark.schema.defaultschema.*;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.*;
+
+public class DefaultSchemaManager extends SchemaManager {
+    private static final Logger LOG = Logger.getLogger(DefaultSchemaManager.class);
+    
+    private Map<String, Table> tables = new HashMap<String, Table>();
+
+    public DefaultSchemaManager(Connection db_connection) {
+        super(db_connection);
+        for (TableSchema t : TPCCTableSchemas.tables.values()) {
+            tables.put(t.name(), new DefaultTable(t));
+        }
+    }
+
+    @Override
+    public void create() throws SQLException {
+        for (Table t : tables.values()) {
+            execute(t.getDropDdl());
+        }
+        for (Table t : tables.values()) {
+            for (String ddl : t.getCreateDdl())
+                execute(ddl);
+        }
+        // TODO -- can we defer this until after load as well?
+        execute("CREATE INDEX idx_customer_name ON customer ((c_w_id,c_d_id) HASH,c_last,c_first)");
+        execute("CREATE UNIQUE INDEX idx_order ON oorder ((o_w_id,o_d_id) HASH,o_c_id,o_id DESC)");
+
+        if (!db_connection.getAutoCommit()) {
+            db_connection.commit();
```

Comment:
Is it really a good place to commit the transaction? This function had not start a transaction, why it should commit something?

Reply:
I'm not really sure. This code was already there so I had left it as it was and hadn't thought too much into it. I could remove it, but is there something I should test for here?